### PR TITLE
Feature component

### PR DESCRIPTION
**Pull request by melkamu12**
### React Capstone Project:

In this PR, I added the initial setup for bookstore.

 ## Activities done
 * [ ] Chose Geonames API for numeric data retrieval based on parameters.
 * [ ] Implemented API key authentication as required.
 * [ ] Followed design guidelines, including color scheme and typography.
 * [ ] Based design on original concept by Nelson Sakwa with proper credit in README.
 ### Home 
* [ ]  Displayed categories with filtering by name.
* [ ] Showed numeric values per category from Geonames API.
* [ ] Enabled navigation to details page upon category selection.

### Details Page:
* [ ] Fetched detailed data from Geonames API for selected category.
* [ ] Implemented "Back" button for navigation.

### Testing:

* [ ] Created unit tests for Redux pure functions.
* [ ] Developed React integration tests for the application using the React Testing Librar.
* [ ] Utilized mocking for API and Redux Store interaction

### Technical Implementation:
* [ ] The project is a single-page application (SPA) built with React and Redu.
* [ ] Data retrieved from the API should be stored in the Redux store.
* [ ] Filter the data retrieve from the API using a Filter stateless componentg.
* [ ] Established unique routes for main and item pages
* [ ] project should be deployed and accessible online using [Render](https://render.com/)

 # Checklist:
 * [ ] Follows the style guidelines of this project
 * [ ] There is no linters error in my code
 * [ ] follow gitflow

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,39 +1,17 @@
...
-  background-color: #282c34;
-  min-height: 100vh;
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  justify-content: center;
-  font-size: calc(10px + 2vmin);
-  color: white;
+body {
+  background-color: rgb(105, 94, 94);
```
Comment (andersonlebon) [inline src/App.css]: - [ ] Kindy  note that the design guidelines need to be followed, make sure your design look the same like the template, pick one color, use the same layout, fontsize, margin, position ....Below you will see the a video where I point more about the design guideline

https://drive.google.com/file/d/1GcEqvu-m97SMtc5Qfsa2S05lcLeks1fV/view?usp=sharing


Your Design | The template 
-----|-----
image image | image image image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @melkamu12 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

almost-there

## Highlights 🍾 

- Nice use of routes and redux 👏  ✔️ 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/CountriesList.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+              countryName={country.countryName}
+              countryCode={country.countryCode}
+              currencyCode={country.currencyCode}
+              areaInSqKm={country.areaInSqKm}
+              continentName={country.continentName}
+              population={country.population}
+            />
+          ))}
+      </ul>
+    </section>
```
Comment (AbayomiOlaoye) [inline src/components/CountriesList.js:33]: - [ ] You have done an impressive job with this implementation, @melkamu12 :clap: 
- The app looks really cool and better but the only issue we currently have is that it not looking so great on the mobile screen. Somehow, the country cards are not equal when the screen gets smaller and there seems so be a behavior that is proving unpredictable: The styles on the header part seems not to be consistent. Kindly check to ensure that it maintains an 100% width in all devices.
- I left a very detailed video for you [here](https://watch.screencastify.com/v/vbo20972RNIRdkZ9voiV), kindly check it out and apply the fix.
_You are almost done with this, and you are doing so well already!_

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @melkamu12 :wave: :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so_close


 ## Highlights
- Good PR message ✅ 
- Great commit messages ✅ 
- No linter error ✅ 
- App is looking really impressive :heavy_check_mark: 

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/CountriesList.js (hunk 2 above)
Comment (melkamu12) [reply src/components/CountriesList.js:33]: I corrected the responsibility and the card height is equal ` min-width: 105px;
  max-width: 100%;
  background-color: var(--color-c9bcd719);
  display: flex;
  height: 150px;`

**Review by KDlamini — APPROVED** [review]
### Hi @melkamu12 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Profesional pull request
✔️ The Home page has a list of categories that could be filtered using given parameters
✔️ The project is a single-page application (SPA) using `react-router-dom`
✔️ The category item detail page has its unique route within the SPA.
✔️ Design guidelines are followed

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**